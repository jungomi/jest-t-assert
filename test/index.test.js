import tTest, { t } from '../src';

tTest('all assertions are successful with t as callback', t => {
  const obj = { start: 'abc', end: 'xyz' };
  t.pass();
  expect(() => t.fail()).toThrow(/t\.fail/);
  t.true(true);
  t.false(false);
  t.truthy(obj);
  t.falsy(0);
  t.is(obj.start, 'abc');
  t.not(obj.start, 'xyz');
  t.deepEqual(obj, { start: 'abc', end: 'xyz' });
  t.notDeepEqual(obj, { start: 'abc', middle: 'lmn' });
  t.throws(() => {
    throw new Error('This function throws');
  });
  t.notThrows(() => {});
  t.regex(obj.start, /bc$/);
  t.notRegex(obj.start, /^bc$/);
  t.snapshot(obj);
  t.plan(15);
});

test('all assertions are successful with the imported t', () => {
  const obj = { start: 'abc', end: 'xyz' };
  t.pass();
  expect(() => t.fail()).toThrow(/t\.fail/);
  t.true(true);
  t.false(false);
  t.truthy(obj);
  t.falsy(0);
  t.is(obj.start, 'abc');
  t.not(obj.start, 'xyz');
  t.deepEqual(obj, { start: 'abc', end: 'xyz' });
  t.notDeepEqual(obj, { start: 'abc', middle: 'lmn' });
  t.throws(() => {
    throw new Error('This function throws');
  });
  t.notThrows(() => {});
  t.regex(obj.start, /bc$/);
  t.notRegex(obj.start, /^bc$/);
  t.snapshot(obj);
  t.plan(15);
});

function delayedMessage(msg, ms) {
  return new Promise(resolve => setTimeout(() => resolve(msg), ms));
}

tTest('returning a promise is handled correctly', t => {
  t.plan(1);
  return delayedMessage('hello', 500).then(t.snapshot);
});

tTest('async/await works correctly', async t => {
  t.plan(1);
  const msg = await delayedMessage('hello', 500);
  t.snapshot(msg);
});
