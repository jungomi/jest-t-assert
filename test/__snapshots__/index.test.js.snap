// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all assertions are successful with t as callback 1`] = `
Object {
  "end": "xyz",
  "start": "abc",
}
`;

exports[`all assertions are successful with the imported t 1`] = `
Object {
  "end": "xyz",
  "start": "abc",
}
`;

exports[`async/await works correctly 1`] = `"hello"`;

exports[`returning a promise is handled correctly 1`] = `"hello"`;
